### ISSUE TITLE
getByRole(role, { current: true })

### ISSUE THREAD

**Issue by josephrexme** [testing-library/dom-testing-library#864]
### Describe the feature you'd like:

The [getByRole query](https://testing-library.com/docs/dom-testing-library/api-queries#byrole) is very versatile and helpful in helping us put accessibility at the forefront of our testing. It currently includes options like `selected`, `checked` to select based on state. This is great but it seem to be missing out on one of such states that is usually defined by users to determine active/current section: [aria-current](https://www.aditus.io/aria/aria-current/). It will be great if we can pass that option that checks for this aria attribute
### Suggested implementation:

Since `aria-current` may be used in different ways e.g `aria-current="page"`, `aria-current="location"`, `aria-current="step"` we may consider making the option take a string like:
```js
getByRole("button", { current: "page" })
```
but so far in my presumed usage it seem like this can be sufficient as:
```js
getByRole("button", { current: true })
```
checking only the existence of the `aria-current` attribute as it's unlikely more than one type is being tested at a time.

**Comment by savcni01** [comment]
I plan to work on this issue today.

**Comment by IanIsFluent** [comment]
This is great! I realise this isn't the right place - but do you know if there'd be any plan for a matcher in `@testing-library/jest-dom` for this?

I'd just rather write `screen.getByText('Page 1 of 3').toHaveAriaCurrent()` than `screen.getByRole('listitem', { current: true }).toHaveTextContent('Page 1 of 3')` 😅

**Comment by terraelise** [comment]
Same as @IanIsFluent neither `expect(screen.getByRole('link', { current: 'page' })).toBeInTheDocument();` nor `expect(getByRole('link', { current: true' })).toBeInTheDocument();` worked, result is always `TestingLibraryElementError: Unable to find an accessible element with the role "link"`
```
link:

      Name "Text Only":
      <a
        class="flex items-center justify-center h-16 px-5 text-body2 font-semibold transition text-neutral3 group-hover:text-neutral1 group-focus:text-neutral1"
        href="#catch-action"
      />
```

**Comment by Kosai106** [comment]
Late to the party @IanIsFluent but based on your code example, it should be `screen.getByRole('listitem', { current: 'step' })`
